src/mappings/rewards.ts

```diff
@@ -8,19 +8,56 @@ import { BigDecimal, BigInt, Address, Bytes,log } from "@graphprotocol/graph-ts"
 
 
 export function handleCurveAaveGaugeUpdate(event: UpdateLiquidityLimit): void {
```

Comment:
I've been thinking about whether we want to consolidate logic by function (e.g. rewards) or by protocol (e.g. mappings/curve.ts). I imagine it's more scalable to consolidate it by protocol - each protocol has its own rewards logic and its own handers. Something to think about for future.

Reply:
It's very close to that already, what is preventing this right now is that the vault each gauge belongs to needs to be defined in this function. If that can be looked up dynamically (which might be possible) or this can be defined in the yaml file (which theGraph has a proposal to allow) it would allow all these methods to be consolidated into one. This should definitely be possible eventually.